Source/WebCore/inspector/InspectorStyleSheet.cpp

```diff
@@ -1141,7 +1145,6 @@ ExceptionOr<void> InspectorStyleSheet::setRuleHeaderText(const InspectorCSSId& i
         // Set the style sheet text directly so we don't rebuild our flat rule set. The CSSStyleRule has been directly
         // updated already.
         m_parsedStyleSheet->setText(sheetText);
-        m_pageStyleSheet->clearHadRulesMutation();
         fireStyleSheetChanged();
```

Comment:
Since the "fast-path" (using CSSOM to update the header text) instead of re-parsing for style rules, we need to restore the `m_pageStyleSheet`'s `wasMutatedByJS` here to not accidentally show the modification indicator in the frontend.